tests/Unit/Domain/Test_SizeOfElement.cpp

```diff
@@ -13,95 +13,228 @@
 #include "Domain/CoordinateMaps/Affine.hpp"
 #include "Domain/CoordinateMaps/CoordinateMap.hpp"
 #include "Domain/CoordinateMaps/CoordinateMap.tpp"
+#include "Domain/CoordinateMaps/CubicScale.hpp"
 #include "Domain/CoordinateMaps/DiscreteRotation.hpp"
+#include "Domain/CoordinateMaps/Identity.hpp"
 #include "Domain/CoordinateMaps/ProductMaps.hpp"
 #include "Domain/CoordinateMaps/ProductMaps.tpp"
 #include "Domain/CoordinateMaps/Rotation.hpp"
 #include "Domain/Direction.hpp"
 #include "Domain/ElementId.hpp"
 #include "Domain/ElementMap.hpp"
+#include "Domain/FunctionsOfTime/FunctionOfTime.hpp"
+#include "Domain/FunctionsOfTime/PiecewisePolynomial.hpp"
 #include "Domain/OrientationMap.hpp"
 #include "Domain/SegmentId.hpp"
 #include "Domain/SizeOfElement.hpp"
 #include "Domain/Tags.hpp"
+#include "Time/Tags.hpp"
 #include "Utilities/MakeArray.hpp"
 #include "Utilities/TMPL.hpp"
 
 namespace {
-void test_1d() noexcept {
+template <size_t Dim>
+using CubicScale = domain::CoordMapsTimeDependent::CubicScale<Dim>;
+
+template <bool UseMovingMesh>
+void test_1d(
+    const std::array<double, 4>& times_to_check,
+    const std::unordered_map<
+        std::string, std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>&
+        functions_of_time) noexcept {
   INFO("1d");
-  auto map = domain::make_coordinate_map_base<Frame::Logical, Frame::Inertial>(
+  auto map = domain::make_coordinate_map_base<Frame::Logical, Frame::Grid>(
       domain::CoordinateMaps::Affine(-1.0, 1.0, 0.3, 1.2));
   const ElementId<1> element_id(0,
                                 std::array<SegmentId, 1>({{SegmentId(2, 3)}}));
-  const ElementMap<1, Frame::Inertial> element_map(element_id, std::move(map));
-  const auto size = size_of_element(element_map);
-  // for this affine map, expected size = width of block / number of elements
-  const auto size_expected = make_array<1>(0.225);
-  CHECK_ITERABLE_APPROX(size, size_expected);
+  const ElementMap<1, Frame::Grid> logical_to_grid_map(element_id,
+                                                       std::move(map));
+
+  const auto check_helper =
+      [&functions_of_time, &logical_to_grid_map,
+       &times_to_check](const auto& grid_to_inertial_map) noexcept {
+        std::array<double, 1> element_size{};
+        for (const double time : times_to_check) {
+          size_of_element(make_not_null(&element_size), logical_to_grid_map,
+                          grid_to_inertial_map, time, functions_of_time);
+          // for this affine map, expected size = width of block / number of
+          // elements
+          const auto expected_size_of_element =
+              make_array<1>(0.225 * (UseMovingMesh ? time : 1.0));
```

Comment:
I think it's somewhat misleading to accept the `functions_of_time` as an argument to the test function, when the test looks like it can only work for one particular function of time. Instead, I suggest adding a helper function like `make_linear_expansion_function_of_time` (I'm sure better names can be had) that you can call within each `test_Nd` function, and then remove the function of time argument.